screed/screedRecord.py

```diff
@@ -180,23 +180,31 @@ def _buildRecord(fieldTuple, dbObj, rowName, queryBy):
     return Record(hackedResult)
 
 
-class _Writer(object):
-
-    def __init__(self, filename, fp=None):
-        self.filename = filename
-        if fp is None:
-            if filename.endswith('.gz'):
-                fp = gzip.open(filename, 'w')
-            elif filename.endswith('.bz2'):
-                fp = bz2.BZ2File(filename, 'w')
-            else:
-                fp = file(filename, 'wb')
-
-        self.fp = fp
-
-    def consume(self, read_iter):
-        for read in read_iter:
-            self.write(read)
-
-    def close(self):
-        self.fp.close()
+def write_fastx(record, fileobj):
+    """Write sequence record to 'fileobj' in FASTA/FASTQ format."""
+    defline = record.name
+    if hasattr(record, 'description'):
+        defline += ' ' + record.description
+
+    if hasattr(record, 'quality'):
+        recstr = '@{defline}\n{sequence}\n+\n{quality}\n'.format(
+            defline=defline,
+            sequence=record.sequence,
+            quality=record.quality)
+    else:
+        recstr = '>{defline}\n{sequence}\n'.format(
+            defline=defline,
+            sequence=record.sequence)
+
+    try:
+        fileobj.write(bytes(recstr, 'utf-8'))
```

Comment:
`f.write(recstr.encode('utf-8'))` should work in both versions of python and get rid of the need for `try ... except`. I think.